homeassistant/components/nws/config_flow.py

```diff
@@ -0,0 +1,90 @@
+"""Config flow for National Weather Service (NWS) integration."""
+import logging
+
+import aiohttp
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from . import NwsData, base_unique_id
+from .const import CONF_STATION, DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    latitude = data[CONF_LATITUDE]
+    longitude = data[CONF_LONGITUDE]
+    api_key = data[CONF_API_KEY]
+    station = data.get(CONF_STATION)
+
+    client_session = async_get_clientsession(hass)
+    ha_api_key = f"{api_key} homeassistant"
+    nws = NwsData(hass, latitude, longitude, ha_api_key, client_session)
+
+    try:
+        await nws.async_set_station(station)
+    except aiohttp.ClientError as err:
+        _LOGGER.error("Could not connect: %s", err)
+        raise CannotConnect
+
+    return {"title": nws.station}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for National Weather Service (NWS)."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            existing_id = await self.async_set_unique_id(
+                base_unique_id(user_input[CONF_LATITUDE], user_input[CONF_LONGITUDE])
+            )
+            if existing_id:
+                errors["base"] = "already_configured"
+            else:
```

Comment:
```suggestion
            await self.async_set_unique_id(base_unique_id(user_input[CONF_LATITUDE], user_input[CONF_LONGITUDE]))
            self._abort_if_unique_id_configured()
```

Reply:
Should this method be used even though it has leading underscore?  Looks much cleaner usage, but it must be marked to be an internal method for a reason.

Reply:
```
# git grep self._abort_if_unique_id_configured|wc
     79     165    5961
```

Yes. Its fine & used in 79 other places in the codebase